src/Gate.php

```diff
@@ -73,4 +90,39 @@ public function handleCallback(string $data): Callback
     {
         return new Callback($data, $this->signatureHandler);
     }
+
+
+    /**
+     * @param Payment $payment
+     * @return void
+     * @throws ValidationException
+     */
+    private function validateParams(Payment $payment)
+    {
+        $requestUri = $this->paymentPageUrlBuilder->getValidationUrl($payment);
+        $stream = fopen($requestUri, 'r');
+        $errors = [];
+        $status = 0;
+
+        // Reverse required!!!
+        $headers = array_reverse(stream_get_meta_data($stream)['wrapper_data']);
+
+        foreach ($headers as $header) {
+            if (preg_match('/^HTTP\/\d.\d (\d+) /', $header, $match)) {
+                $status = (int) $match[1];
+                break;
+            }
+        }
+
+        if ($status !== 200) {
+            $data = json_decode(stream_get_contents($stream));
```

Comment:
Can we make throw new ValidationException in this place? After than, remove unnecessary variables.

Reply:
No, we can not do that, because we need to close stream before throw exception.